### PR TITLE
[docs] clarify description of prediction early stopping

### PR DESCRIPTION
**Pull request by StrikerRUS**
https://github.com/microsoft/LightGBM/blob/f83180883a51729af9562ccac7b450c6da79a04a/src/application/predictor.hpp#L43-L58

https://github.com/microsoft/LightGBM/blob/d517ba12f2e7862ac533908304dddbd770655d2b/include/LightGBM/boosting.h#L282-L283

https://github.com/microsoft/LightGBM/blob/8116d880f79967d339783e5e9597351639232015/include/LightGBM/objective_function.h#L61-L62

### CODE REVIEW

File: docs/Parameters.rst [hunk 1]
```diff
@@ -909,8 +909,12 @@ Predict Parameters
 
    -  used only in ``prediction`` task
 
+   -  used only in ``classification`` and ``ranking`` applications
```
Comment (StrikerRUS) [inline docs/Parameters.rst:912]: ### Classification
https://github.com/microsoft/LightGBM/blob/0701a32da9fae7b2de8a01d702fa7d6abf36e836/src/objective/binary_objective.hpp#L188
https://github.com/microsoft/LightGBM/blob/d517ba12f2e7862ac533908304dddbd770655d2b/src/objective/multiclass_objective.hpp#L153
https://github.com/microsoft/LightGBM/blob/d517ba12f2e7862ac533908304dddbd770655d2b/src/objective/multiclass_objective.hpp#L259

### Ranking
https://github.com/microsoft/LightGBM/blob/8116d880f79967d339783e5e9597351639232015/src/objective/rank_objective.hpp#L80

File: docs/Parameters.rst [hunk 2]
```diff
@@ -909,8 +909,12 @@ Predict Parameters
...
+
    -  if ``true``, will use early-stopping to speed up the prediction. May affect the accuracy
 
+   -  **Note**: cannot be used with ``rf`` boosting type or custom objective function
```
Comment (StrikerRUS) [inline docs/Parameters.rst:916]: https://github.com/microsoft/LightGBM/blob/8116d880f79967d339783e5e9597351639232015/src/boosting/rf.hpp#L205-L208
https://github.com/microsoft/LightGBM/blob/a8ee487aca35363fafa027e6b7695976045096b3/src/boosting/gbdt.h#L159-L169

**Comment by StrikerRUS** [comment]
Does prediction early stopping really work with ranking applications?

**Review by jameslamb — APPROVED** [review]
Looks good to me based on the links provided, thanks for doing that!

I don't know the answer to https://github.com/microsoft/LightGBM/pull/4411#issuecomment-869059869 though.

**Comment by StrikerRUS** [comment]
@cbecker @guolinke Could you please help with this question https://github.com/microsoft/LightGBM/pull/4411#issuecomment-869059869?

**Comment by cbecker** [comment]
@StrikerRUS It depends on what the scores look for ranking.
For multi-class, each instance is given N scores [c1, c2, c3 c4, .. cn], and we compute the margin (see `prediction_early_stop.cpp`, `CreateMulticlass()`). The idea is that when a class is further than all other by a large margin (score difference), then we can stop.

I see that for ranking LightGBM implements regression. In this case, what is the prediction for each sample?
If it is a relative 'rank' to other values as a difference, as in multi-class, then one could apply the multi-class early stopping computation above (though I never tried that in practice)

**Comment by StrikerRUS** [comment]
@cbecker Thank you so much for quick response!
LightGBM's prediction values in ranking are pointwise (https://github.com/microsoft/LightGBM/issues/3326#issuecomment-677957226).

**Pull request merged**